### ISSUE TITLE
The shuffle operation in the data loading process.

### ISSUE THREAD

**Issue by wanghao14** [vt-vl-lab/DF-Net#22]
Hi, thanks for your inspiring work! I am new to tensorflow and optical flow estimation, and I am trying to learn your code recently. I don't quite understand [here](https://github.com/vt-vl-lab/DF-Net/blob/master/core/data_loader.py#L30) in your code: the order of the data sequence is shuffled. In this way, can we ensure that a batch of images read from the sequence is consecutive? 
Because tensorflow has few documents, I don't know if my understanding is correct. Looking forward to your response.

**Comment by Yuliang-Zou** [comment]
Hi @wanghao14 , actually the consecutive frames are concatenated together during data preprocessing. So we can make sure we can get consecutive frames even if the data is shuffled.

**Comment by wanghao14** [comment]
@Yuliang-Zou Sorry, I still have some questions. 
In the function named [format_file_list](https://github.com/vt-vl-lab/DF-Net/blob/53f4e016b881d55624042f755235eb8d7d248209/core/data_loader.py#L173), the path to the images is stored in the list file_list['image_file_list'], which looks like 

> ['$data_root$/2011_09_26_drive_0001_sync_02/0000000002.jpg', '$data_root$/2011_09_26_drive_0001_sync_02/0000000004.jpg', '$data_root$/2011_09_26_drive_0001_sync_02/0000000005.jpg',..., '$data_root$/2011_10_03_drive_0042_sync_03/0000001166.jpg']

.
Then, it will be used as the input to [tf.train.string_input_producer with shuffling](https://github.com/vt-vl-lab/DF-Net/blob/53f4e016b881d55624042f755235eb8d7d248209/core/data_loader.py#L27), and I think the sequence looks like 

> ['$data_root$/2011_09_26_drive_0001_sync_02/0000000005.jpg', '$data_root$/2011_10_03_drive_0042_sync_03/0000001066.jpg', '$data_root$/2011_09_30_drive_0028_sync_03/0000002128.jpg', '$data_root$/2011_09_26_drive_0001_sync_02/0000000036.jpg', ...]

, which means that the order of image sequence is not consecutive. Then in the function named [unpack_image_sequence](https://github.com/vt-vl-lab/DF-Net/blob/53f4e016b881d55624042f755235eb8d7d248209/core/data_loader.py#L187), the middle(3th) image is treated as target image and the first two images and last two images are treated as source images. I don't think these five images are consecutive, and this is where I am confused. Could you please explain this part? Or some details of your data preprocessing? Because I am new to tensorflow, my question may be naive. I hope it doesn't bother you.

**Comment by Yuliang-Zou** [comment]
You can see the data preprocessing in `data/kitti/kitti_raw_loader.py`. We actually concatenate consecutive frames before we start training.
